src/test/result_tests.cpp

```diff
@@ -63,6 +63,7 @@ void ExpectSuccess(const util::Result<T>& result, const bilingual_str& str, Args
 {
     ExpectResult(result, true, str);
     BOOST_CHECK_EQUAL(result.has_value(), true);
+    // NOLINTNEXTLINE(bugprone-use-after-move)
```

Comment:
Right. I outlined this issue for follow-ups in the PR description.

Reply:
We can likely simplify the call here, see: https://github.com/bitcoin/bitcoin/pull/31306#discussion_r1846343700

Reply:
Fixing all new "bugprone-use-after-move" warnings contradicts with other [comments](https://github.com/bitcoin/bitcoin/pull/31306#discussion_r1846324884). On the other hand, fixing some of them is not consistent. That's why I chose to silence them.

`NOLINT` comments are easy searchable and effectively document the issues that have to be resolved.

Reply:
Unfortunately silencing will also invalidate the ACK-ed PR.
But if we could instead add `-bugprone-use-after-move` to `src/.clang-tidy` this concern could be addressed separately.

Reply:
It will expose the whole codebase to re-introducing related bugs, no?

Reply:
We could tackle it in a separate pr like we do with the `performance-*` checks, right?

Reply:
We could, but `performance-*` checks are not about safety, whereas `bugprone-*` checks are.

Reply:
If you decide to fix them here, I'll just rebase the related PR and ask for a re-review, it's not a tragedy :)

Reply:
> NOLINT comments are easy searchable and effectively document the issues that have to be resolved.

Sorry, but no, that is exactly the _opposite_ if what `NOLINT` documents!

`NOLINT` means "I'm smarter than my static analyzer".

`TODO` would effectively document an issue to be resolved.

Hard NACK here due to the reasoning being applied. In fact, it rises to the level of suspicion. I'll be reviewing all of these refactors in great detail.

Reply:
@hebasto, what if we invert this and push the fixes in separate PRs and only merge this TIDY_LLVM_V upgrade after all problems were already resolved?

Reply:
> > NOLINT comments are easy searchable and effectively document the issues that have to be resolved.
> 
> Sorry, but no, that is exactly the _opposite_ of what `NOLINT` documents!
> 
> `NOLINT` means "I'm smarter than my static analyzer".
> 
> `TODO` would effectively document an issue to be resolved.
> 
> Hard NACK here due to the reasoning being applied. In fact, it rises to the level of suspicion. I'll be reviewing all of these refactors in great detail.

A bare `NOLINT`, without any explanation, has never meant "I'm smarter than my static analyzer"—at least, not to me.

Reply:
> A bare NOLINT, without any explanation, has never meant "I'm smarter than my static analyzer"—at least, not to me.

I'm not sure what else it could mean.

[From the docs](https://clang.llvm.org/extra/clang-tidy/): "clang-tidy diagnostics are intended to call out code that does not adhere to a coding standard, or is otherwise problematic in some way. However, if the code is known to be correct, it may be useful to silence the warning"

Does this mean there are other places in the code where `NOLINT` is used as a means of documenting a todo? If so, we need to address those asap.

Reply:
> I'm not sure what else it could mean.
> 
> [From the docs](https://clang.llvm.org/extra/clang-tidy/): "clang-tidy diagnostics are intended to call out code that does not adhere to a coding standard, or is otherwise problematic in some way. However, if the code is known to be correct, it may be useful to silence the warning"

You are right.

Reply:
@hebasto No need to close this if we can change the approach. I think the changes are useful (once the actual bugs are fixed), but your suggestion that other real bugs might have been previously masked set off my alarm bells. Looking through the code now (bench/tests excluded), I don't see any other examples of this.

I think we can agree on an approach where we (in this order):
- Fix the actual bugs
- Notate the false-positives
- Enable the checks.

It sounds like it also is worth adding a line in `developer-notes.md` that describes when a `NOLINT` is/is not appropriate.

Reply:
> Have you seen this suggestion?

Thanks! Your suggestion has been accepted.